Nancy/quickstart edits (#26)
ShiftLeftSecurity/documentation: using-inspect-protect/quick-start.md

The process for quickly starting with ShiftLeft Inspect and ShiftLeft Protect is:

1. [Choose your operating system](#supported-operating-systems).
   
2. [Install the ShiftLeft Command Line Interface (CLI)](#installing-the-shiftleft-cli).

3. [Authenticate with ShiftLeft](#authenticating-with-shiftleft).

4. [Run ShiftLeft Protect](#running-shiftleft-protect).

The information in this article parallels and provides additional details on the Quick Start page that you use when you first log into ShiftLeft. 

## Supported Operating Systems

ShiftLeft supports the Linux, MacOS X, Windows operating systems.

## Installing the ShiftLeft CLI

The ShiftLeft (CLI) is used with ShiftLeft Inspect to submit applications for analysis and with ShiftLeft Protect to run the ShiftLeft Microagent. The tool is named sl.

There are two methods for installing the ShiftLeft CLI: [downloading the installer](#downloading-the-shiftleft-cli-installer) or [running the appropriate CLI install command](#running-the-cli-install-command).

### Downloading the ShiftLeft CLI Installer

To download the ShiftLeft CLI Installer:

1. From the Quick Start page, click the Download the ShiftLeft CLI button.

> ![](download-cli.jpg)

Author: Done

2. Add the CLI to your system path. The process is different for each operating system, as explained below.

3. Verify the CLI installation by typing `sl help`. 

See more information on [Using the ShiftLeft CLI](using-sl-the-shiftleft-cli.md).

#### Linux and MacOS X

Make sure that /usr/local/bin is in your $PATH.

Extract the sl/sl.exe binary and then add the directory location of the sl binary to your $PATH (or manually copy it to /usr/local/bin).

#### Windows .Net

Note that for Windows .NET there are two variants: .NET Framework or .NET Core version. Make sure to pick the right one for your project.

After you have downloaded the appropriate installer, unzip the file and invoke the installer. If you are running the installer from the terminal, add '--no-prompt' to disable waiting for user input.

### Running the CLI Install Command

To run the CLI Install command:

* From the Quick Start page, copy the command and run it from your command line. The command displayed is appropriate to your chosen operating system.

> ![](run-install-command.jpg)

Author: Done

You can also copy the command from here:

#### Linux

> curl https://cdn.shiftleft.io/download/sl-latest-linux-x64.tar.gz | tar xvz -C /usr/local/bin

Reviewer: This should go in a code block.

Author: Done

#### MacOS X

curl https://cdn.shiftleft.io/download/sl-latest-osx-x64.tar.gz | tar xvz -C /usr/local/bin

#### Windows .NET Framework

Invoke-WebRequest -Uri https://cdn.shiftleft.io/download/installer-dotnet-framework-latest-windows-x64.zip -UseBasicParsing -OutFile sl-latest-windows-x64.zip

#### Windows .NET Core

Invoke-WebRequest -Uri https://cdn.shiftleft.io/download/installer-dotnet-core-latest-windows-x64.zip -UseBasicParsing -OutFile sl-latest-windows-x64.zip

## Authenticating with ShiftLeft

To authenticate with ShiftLeft:

* From the Quick Start page, copy the command and run it from your command line. The command automatically includes your unique organization ID and upload token.

> ![](associate.jpg)

Reviewer: Missing text for this image.

Author: Remember you told me to remove the image because it had my unique Org ID and Token?

To analyze the code before running your application, refer to [Analyzing Applications in CI](analyzing-applications-in-ci.md). Note that in this case, you need to carry the produced `shiftleft.json` file to your runtime environment and make it available to the ShiftLeft Protect Microagent. This allows the Microagent to be associated with your application.

To analyze the code during runtime and in at the command line, first verify that you have met the [ShiftLeft Inspect requirements for .NET](../introduction/requirements.md).  Then follow the instructions for analysis using the [ShiftLeft CLI](analyzing-applications-in-ci.md). The system submits your application to the cloud for analysis and waits for the analysis to finish before running the application

### Running ShiftLeft Protect for Java

In order to start your application with ShiftLeft Protect, you need to prefix the command line to start your application with 'sl run', specifically

sl run \
  --app <name>\
  --analyze target/<jar> \

>   -- java -jar target/<delimits flags>

Reviewer: Missing code block

Author: Done

# Quick Start for ShiftLeft Inspect and ShiftLeft Protect

You use ShiftLeft Inspect to analyze your application for vulnerabilites, and ShiftLeft Protect to monitor and protect your application at runtime. To use ShiftLeft, you need:

* A ShiftLeft Account ([contact us](https://www.shiftleft.io/contact/)).
* All [ShiftLeft requirements](../introduction/requirements.md).
* A Java application you want to analyze and protect. You can also use the [HelloShiftLeft demo Java application](https://github.com/ShiftLeftSecurity/HelloShiftLeft) to learn how to use ShiftLeft. 

The process for quickly starting with ShiftLeft Inspect and ShiftLeft Protect is:

1. [Choose your operating system](#supported-operating-systems).
   
2. [Install the ShiftLeft Command Line Interface (CLI)](#installing-the-shiftleft-cli).

3. [Authenticate with ShiftLeft](#authenticating-with-shiftleft).

> 4. [Run ShiftLeft Protect](#running-shiftleft-protect).

Reviewer: As we discussed, there should be another step between authenticating and running (the analysis step) that we might want to talk about. But it's ok to do this as a follow-up PR.

Author: Planned for a follow up PR. I think we need more of a discussion, and include Etan.

The information in this article parallels and provides additional details on the Quick Start page that you use when you first log into ShiftLeft. 

## Supported Operating Systems

> ShiftLeft supports the Linux, MacOS X, Windows operating systems.

Reviewer: ```suggestion
ShiftLeft supports the Linux, MacOS X and Windows operating systems.
```

Author: Done

## Installing the ShiftLeft CLI

The ShiftLeft (CLI) is used with ShiftLeft Inspect to submit applications for analysis and with ShiftLeft Protect to run the ShiftLeft Microagent. The tool is named sl.

There are two methods for installing the ShiftLeft CLI: [downloading the installer](#downloading-the-shiftleft-cli-installer) or [running the appropriate CLI install command](#running-the-cli-install-command).

### Downloading the ShiftLeft CLI Installer

To download the ShiftLeft CLI Installer:

1. From the Quick Start page, click the Download the ShiftLeft CLI button.

![Download the ShiftLeft CLI](download-cli.jpg)

2. Add the CLI to your system path. The process is different for each operating system, as explained below.

3. Verify the CLI installation by typing `sl help`. 

See more information on [Using the ShiftLeft CLI](using-sl-the-shiftleft-cli.md).

#### Linux and MacOS X

> Make sure that /usr/local/bin is in your $PATH.

Reviewer: We should put all such paths or variables between single back-ticks so that they are formatted nicely.

Author: Done. I will make that a standard going forward.

> Make sure that /usr/local/bin is in your $PATH.

Reviewer: ```suggestion
Make sure that `/usr/local/bin` is in your `$PATH`.
```

Author: Done

> Extract the sl/sl.exe binary and then add the directory location of the sl binary to your $PATH (or manually copy it to /usr/local/bin).

Reviewer: Same here and other places.

Author: Done

#### Windows .Net

Note that for Windows .NET there are two variants: .NET Framework or .NET Core version. Make sure to pick the right one for your project.

> After you have downloaded the appropriate installer, unzip the file and invoke the installer. If you are running the installer from the terminal, add '--no-prompt' to disable waiting for user input.

Reviewer: ```suggestion
After you have downloaded the appropriate installer, unzip the file and invoke the installer. If you are running the installer from the terminal, add `--no-prompt` to disable waiting for user input.
```

Author: Done

### Running the CLI Install Command

To run the CLI Install command:

* From the Quick Start page, copy the command and run it from your command line. The command displayed is appropriate to your chosen operating system.

![Command to Run the CLI Installation](run-install-command.jpg)

You can also copy the command from here:

#### Linux

```
curl https://cdn.shiftleft.io/download/sl-latest-linux-x64.tar.gz | tar xvz -C /usr/local/bin
```

#### MacOS X

```
curl https://cdn.shiftleft.io/download/sl-latest-osx-x64.tar.gz | tar xvz -C /usr/local/bin
```

#### Windows .NET Framework

> Invoke-WebRequest -Uri https://cdn.shiftleft.io/download/installer-dotnet-framework-latest-windows-x64.zip -UseBasicParsing -OutFile sl-latest-windows-x64.zip

Reviewer: This should also be in a code block

Author: Done

To analyze the code before running your application, refer to [Analyzing Applications in CI](analyzing-applications-in-ci.md). Note that in this case, you need to carry the produced `shiftleft.json` file to your runtime environment and make it available to the ShiftLeft Protect Microagent. This allows the Microagent to be associated with your application.

To analyze the code during runtime and in at the command line, first verify that you have met the [ShiftLeft Inspect requirements for .NET](../introduction/requirements.md).  Then follow the instructions for analysis using the [ShiftLeft CLI](analyzing-applications-in-ci.md). The system submits your application to the cloud for analysis and waits for the analysis to finish before running the application

### Running ShiftLeft Protect for Java

In order to start your application with ShiftLeft Protect, you need to prefix the command line to start your application with 'sl run', specifically

```
sl run \
  --app <name>\
  --analyze target/<jar> \
  -- java -jar target/<delimits flags>
```
   
where

> --app <name> specifies a unique name for your application.

Reviewer: ```suggestion
* `--app <name>` specifies a unique name for your application.
```

Reviewer: These should be formatted as bullets, otherwise Markdown will put them on all one line.

Author: Done

Once your application is running with ShiftLeft Protect, you can trigger some activity or expose it to real traffic. This activity is shown on the ShiftLeft Dashboard.

For additional information, refer to [Installing ShiftLeft Protect and Configuring the Microagent for Java](../installing-the-microagent/installing-protect-java.md).

### Running ShiftLeft Protect for .Net

To run ShiftLeft Protect .Net, be sure that you:

- Are running on a Windows operating system that has at least .NET Framework Runtime 4.5 installed.
- Analyze the application beforehand [using the ShiftLeft CLI](using-sl-the-shiftleft-cli.md). 

If you are running ShiftLeft Protect with *Internet Information Services (IIS)* you should first copy the `shiftleft.json` file (that was generated during analysis) into the `C:\shiftleftDotNetAgent\spr`directory. You can then start sending requests to your application in the usual manner (e.g., through a browser or a script). The ShiftLeft Protect Microagent is activated once your application receives its first request.

If you are running ShiftLeft Protect with *IIS Express*, start the application through the ShiftLeft CLI as follows:
          

> "C:\Program Files\Shiftleft\sl.exe" run --csharp "C:\Program Files\IIS Express\iisexpress.exe" <full path to the project location>

Reviewer: Needs code block

Author: Done
